clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -4120,6 +4127,9 @@ void CodeGenModule::emitMultiVersionFunctions() {
                                    TA->getArchitecture(), Feats);
             } else {
               const auto *TVA = CurFD->getAttr<TargetVersionAttr>();
+              if (TVA->isDefaultVersion() &&
+                  CurFD->doesThisDeclarationHaveABody())
+                EmitResolver = true;
               llvm::SmallVector<StringRef, 8> Feats;
               TVA->getFeatures(Feats);
               Options.emplace_back(cast<llvm::Function>(Func),
```

Comment:
github won't let me put a comment on the line I want to, but should there be a `EmitResolver = true` down in `} else if (FD->isTargetClonesMultiVersion()) {` ?

Reply:
It is initialized with `!FD->isTargetVersionMultiVersion()` so it is set to true for target clones. However I've just found a bug when we don't have a default definition but we do have a caller in the TU. In that case we should be checking `isUsed()` too. I'll fix this.

Reply:
oh, ok. cool.

Reply:
I always get `isTargetMultiVersion` confused w/ TargetVersionMultiVersion.